src/main/java/org/elasticsearch/index/recovery/RecoveryStats.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.index.recovery;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Streamable;
+import org.elasticsearch.common.unit.TimeValue;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentBuilderString;
+
+import java.io.IOException;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ *
+ */
+public class RecoveryStats implements ToXContent, Streamable {
+
+    private final AtomicInteger currentAsSource = new AtomicInteger();
+    private final AtomicInteger currentAsTarget = new AtomicInteger();
+    private final AtomicLong throttleTimeInNanos = new AtomicLong();
+
+    public RecoveryStats() {
+    }
+
+    public void add(RecoveryStats recoveryStats) {
+        if (recoveryStats == null) {
+            return;
+        }
+        this.currentAsSource.addAndGet(recoveryStats.currentAsSource());
+        this.currentAsTarget.addAndGet(recoveryStats.currentAsTarget());
+        this.throttleTimeInNanos.addAndGet(recoveryStats.throttleTime().nanos());
+    }
+
+    /**
+     * Number of ongoing recoveries for which a shard serves as a source
+     */
+    public int currentAsSource() {
+        return currentAsSource.get();
+    }
+
+    /**
+     * Number of ongoing recoveries for which a shard serves as a source
+     */
+    public int currentAsTarget() {
+        return currentAsTarget.get();
+    }
+
+    /**
+     * Total time recoveries waited due to throttling
+     */
+    public TimeValue throttleTime() {
+        return TimeValue.timeValueNanos(throttleTimeInNanos.get());
+    }
+
+    public void incrementRecoveriesAsTarget() {
+        currentAsTarget.incrementAndGet();
+    }
+
+    public void decrementRecoveriesAsTarget() {
+        currentAsTarget.decrementAndGet();
+    }
+
+    public void incrementRecoveriesAsSource() {
```

Comment:
can we fix the naming here somehow? I had a hard time to get what it means maybe something like `incCurrentSourceRecoveries`?